pkg/endpoint/endpoint.go

```diff
@@ -983,24 +983,21 @@ func (e *Endpoint) LogStatus(typ StatusType, code StatusCode, msg string) {
 	// log message to a single writer?
 	e.Status.indexMU.Lock()
 	defer e.Status.indexMU.Unlock()
-	sts := &statusLogMsg{
-		Status: Status{
-			Code: code,
-			Msg:  msg,
-			Type: typ,
-		},
-		Timestamp: time.Now().UTC(),
-	}
-	e.Status.addStatusLog(sts)
+	e.logStatusLocked(typ, code, msg)
 }
 
 func (e *Endpoint) LogStatusOK(typ StatusType, msg string) {
-	e.Mutex.Lock()
-	defer e.Mutex.Unlock()
-	e.Status.indexMU.Lock()
-	defer e.Status.indexMU.Unlock()
+	e.LogStatus(typ, OK, msg)
+}
+
+func (e *Endpoint) logStatusLocked(typ StatusType, code StatusCode, msg string) {
 	sts := &statusLogMsg{
-		Status:    NewStatusOK(typ, msg),
+		Status: Status{
+			Code:  code,
+			Msg:   msg,
+			Type:  typ,
+			State: e.state,
+		},
 		Timestamp: time.Now().UTC(),
```

Comment:
I realise this is being refactored, but it'd be nice to standardize timestamp output to RFC3339 format - should be time.Now().UTC().Format(time.RFC3339)

Reply:
Ah, good. I wasn't observing any different, just that some of the usages seem to explicitly format to RFC3339, while some don't (just using 'grep' across the tree for similar usages).